test/e2e/upgrade_hypershift_operator_test.go

```diff
@@ -0,0 +1,226 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/cmd/install"
+	"github.com/openshift/hypershift/hypershift-operator/controllers/manifests"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	"sigs.k8s.io/cluster-api/api/v1beta1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	nodePoolAnnotationCurrentConfig        = "hypershift.openshift.io/nodePoolCurrentConfig"
+	nodePoolAnnotationCurrentConfigVersion = "hypershift.openshift.io/nodePoolCurrentConfigVersion"
+	hostedClusterUpgradeTestLabel          = "hypershift.openshift.io/upgrade-test"
+)
+
+// TestUpgradeHyperShiftOperator validates that a HyperShift Operator upgrade won't unnecessarily cause a node rollout.
+// The test must be triggered via the flag upgrade.run-tests and the HyperShift Operator initial image must be set
+// via the flag e2e.hypershift-operator-initial-image.
+func TestUpgradeHyperShiftOperator(t *testing.T) {
+	if !globalOpts.RunUpgradeTest {
+		// This test should be triggered intentionally by setting the option above,
+		// otherwise skip it.
+		t.SkipNow()
+	} else {
```

Comment:
nit: we can drop the `else {}` here, `t.skipNow()` will stop execution immediately.

Reply:
Fixed